securedrop/static/js/source.js

```diff
@@ -124,9 +124,9 @@ function suggestTor() {
 
 function checkClearnet() {
   let url = new URL(location.href);
-  // Allow localhost for development
-  let localhost = ["127.0.0.1", "localhost"];
-  if (localhost.indexOf(url.hostname) === -1 && !url.host.endsWith(".onion")) {
+  // Allow localhost for development, and official demo hostnames for demo
```

Comment:
Can we do what the demo does currently, and patch in these hostnames rather than putting them in the source that goes out to all SecureDrops? I think it would be relatively straightforward to do when building the container?

Reply:
We'd need to wrap the build process somehow, which is difficult with the way we currently build web containers in CI (it uses something declarative instead of directly running a command that runs `docker build`).

If including this check in the code served to the client could depend on something in the server environment, that'd be easy, but it's a static file :-/ .

I can definitely remove the `staging-` versions, if the concern is that those are not publicly advertised (they're going to be torn down anyway).

Reply:
(I did go ahead and do that, so there's only one entry added.)

Reply:
What I had in mind was using https://github.com/freedomofpress/infrastructure/blob/70ee349d3e5f0b570b8444d181768f4151835145/roles/sd-demo/files/0001-Disable-tor2web-redirect-for-localization-demo.patch, copying it into the container, and then applying it to `/opt/securedrop/` before continuing with the build process.

Even simpler would be `sed -i 's/"localhost"\];/"localhost", "demo-source.securedrop.org"];/' /opt/securedrop/securedrop/static/js/source.js`

I'm mostly proposing this since that what Conor had done earlier. But if you think we should just hardcode it in the source, I'm okay with that too if @zenmonkeykstop is onboard too.

Reply:
My pref would be the patch route if possible, just to keep the code shipped to prod clean.

Reply:
OK, I've switched it to sed in the Dockerfile.